front/app/containers/App/index.tsx

```diff
@@ -97,29 +95,32 @@ const App = ({ children }: Props) => {
   ] = useState(false);
   const [userSuccessfullyDeleted, setUserSuccessfullyDeleted] = useState(false);
 
-  const [locale, setLocale] = useState<SupportedLocale | null>(null);
-
   const redirectsEnabled = useFeatureFlag({ name: 'redirects' });
 
+  useEffect(() => {
+    const momentLocale = appLocalesMomentPairs[locale] || 'en';
+    moment.locale(momentLocale);
+  }, [locale]);
+
   useEffect(() => {
     if (appConfiguration && !isAppInitialized) {
       // Set the default timezone
       moment.tz.setDefault(
         appConfiguration.data.attributes.settings.core.timezone
       );
 
-      uniq(
+      const appConfigMomentLocales = uniq(
         appConfiguration.data.attributes.settings.core.locales
           .filter((loc) => loc !== 'en')
           .map((loc) => appLocalesMomentPairs[loc])
-      ).map(async (loc) => {
+      );
+
+      appConfigMomentLocales.forEach(async (momentLocale) => {
```

Comment:
It makes sense to import the files for all appConfig locales. We could look at delaying some imports, but wouldn't do that now.